src/mtev_http.c

```diff
@@ -981,6 +1008,13 @@ mtev_http_complete_request(mtev_http_session_ctx *ctx, int mask) {
     if(len <= 0) goto full_error;
     if(len > 0) in->size += len;
     rv = mtev_http_request_finalize(&ctx->req, &err);
+    /* walk the bchain and set the compression */
+    struct bchain *x = ctx->req.first_input;
+    while (x) {
+      x->compression = request_compression_type(&ctx->req);
+      ctx->req.content_length_read += x->size;
```

Comment:
Reading this, another thing occurs to me - that there might be multiple requests chained together on the connection. That is, the bchain could include something like ```
GET /state HTTP=1.1

GET /something/else HTTP=1.1

```
Probably not very likely, I don't think it's a showstopper for the merge.